File: cowork/Domain/Cowork/TimeSlot.cs
using System;

namespace coworkdomain.Cowork {

    public class TimeSlot {

        public TimeSlot(long id, DayOfWeek day, short startHour, short startMinutes, short endHour, short endMinutes, long placeId) {
            Id = id;
            Day = day;
            StartHour = startHour;
            StartMinutes = startMinutes;
            EndHour = endHour;
            EndMinutes = endMinutes;
            PlaceId = placeId;
        }
        
        public  TimeSlot() {}

        public long Id { get; set; }
        public short StartHour { get; set; }
        public short StartMinutes { get; set; }
        public short EndHour { get; set; }
        public short EndMinutes { get; set; }
        public long PlaceId { get; set; }
        public Place Place { get; set; }
        public DayOfWeek Day { get; set; }

    }

}

File: cowork/ClientApp/src/app/models/TimeSlot.ts
import {WeekDay} from '@angular/common';
import {Place} from './Place';
import {List} from 'linqts';

export class TimeSlot {
    public Id: number;
    public StartHour: number;
    public StartMinutes: number;
    public EndHour: number;
    public EndMinutes: number;
    public PlaceId: number;
    public Place: Place;
    public Day: WeekDay;
    
    public static OrderByDay(slots: TimeSlot[]) {
        let list = new List<TimeSlot>(slots);
        if(list.Count() === 0) return list.ToArray(); 
        return list.Where(sl => sl.Day != 0).OrderBy(slot => slot.Day).ToArray();
    }
}
